pkg/channel/event_dispatcher.go

```diff
@@ -71,30 +73,32 @@ func NewEventDispatcher(logger *zap.Logger) *EventDispatcher {
 // DispatchEvent dispatches an event to a destination over HTTP.
 //
 // The destination and reply are URLs.
-func (d *EventDispatcher) DispatchEvent(ctx context.Context, event cloudevents.Event, destination, reply string) error {
+func (d *EventDispatcher) DispatchEvent(ctx context.Context, event cloudevents.Event, destination, reply *apis.URL) error {
 	var err error
 	// Default to replying with the original event. If there is a destination, then replace it
 	// with the response from the call to the destination instead.
 	response := &event
-	if destination != "" {
-		destinationURL := d.resolveURL(destination)
-		ctx, response, err = d.executeRequest(ctx, destinationURL, event)
+	if destination != nil {
```

Comment:
You might want to check here that `destination.Scheme != ""`, since we've said that's nonsensical (and below on `reply`).

This also helps preserve the current behavior of the code if you change the argument types from `*apis.URL` to `apis.URL`.